controllers/bucket_controller.go

```diff
@@ -188,3 +203,26 @@ func removeKey(slice []string, s string) []string {
 	}
 	return slice
 }
+
+func (b *BucketReconciler) WaitForSecret(namespace, name string) (*corev1.Secret, error) {
+	// set a timeout of 10 minutes
+	timeout := 10 * time.Minute
+
+	secret := corev1.Secret{}
+	wait.PollImmediate(5*time.Second, timeout, func() (bool, error) {
+		key := types.NamespacedName{
```

Comment:
each 5 seconds this var is created? should we put it outside func body?

Should we check if `wait.PollImmediate` returned an error? The way it is, func always return `nil` as error, right? (or just return `*corev1.Secret` and not error as well?)

Reply:
Done, updated the PR.